### ISSUE TITLE
Notebook link with ID

### ISSUE THREAD

**Issue by Cletip** [orgzly/orgzly-android#939]
I search to make a link to a notebook with a id. The notebook is like that : 

:PROPERTIES:
:ID:       9c8b2e3a-3206-4de3-890f-dd91fbba9ebf
:END:

Some texte or title.

This would allow to browse files of org-roam in orgzly, which would be really useful (not see the backlinks, just browse files).
The functionality may already exist, but I can't find it :/. 
Thanks in advance for your answers

**Comment by markokocic** [comment]
Looks like id links work already for tasks/notes, but not for files/notebooks.

**Comment by Cletip** [comment]
> Looks like id links work already for tasks/notes, but not for files/notebooks. 

Exactly what I found. The "link with a file having an ID" functionality therefore does not yet exist

**Comment by artyhedgehog** [comment]
Would be really useful to support [org-roam](https://www.orgroam.com/).

I'm thinking about contributing for this feature, if it is potentially possible to be merged (or otherwise to just fork it for my personal needs). I don't have experience in mobile development, but I do have personal need for the feature, therefore a motivation.

Can you guys give me some leads for existing ID-related functionality code? Probably need to start from a place where Orgzly parses ID from tasks for indexing? 

Also might be necessary to support nested directory, as org-roam usually stores a note per file in its own dir inside the main Orgzly directory. E.g.:
```
+ org/
++ index.org
++ tasks.org
++ ideas.org
++ roam/
+++ 202207041952-a-single-note-with-its-own-id.org
+++ 202207051709-another-note.org
```

**Comment by Cletip** [comment]
> Can you guys give me some leads for existing ID-related functionality code? 

I don't know the exact meaning of this sentence, but if you are looking for "the features wanted for this feature", yes I can give you some:

- must have, for the user, the same functionality as the other links 
- be able to retrieve the id, and search for an associated notebook...
- ... It doesn't matter where this notebook is stored (I'm just talking about the different folders where orgzly looks, not on the whole phone

> Probably need to start from a place where Orgzly parses ID from tasks for indexing?

Yes, this may be a good idea. I was also thinking of something else (I didn't look at the code in detail afterwards):
I think that headings are objects that are stored in a database. They have an "id" field. When the user taps on a link with an id, a query has to be made in the database. If it finds a heading, it teleports to that heading. 
Why not do the same thing for a notebook? Simply add an "id" field that would be in the database, and simply update it with a regular expression that checks if the id has changed when saving the notebook?

Then, when the user clicks on a link, makes the search (= query to the database) by searching also in the Notebook object the id.

I didn't look at the code, maybe it's a crazy idea.

> Also might be necessary to support nested directory (related issue: #926), as org-roam usually stores a note per file in its own dir inside the main Orgzly directory. 

I think that directories should be managed by orgzly. You have to add by hand all the folders, an option would be convenient to look for nested folders, but I think it should be managed by the user in orgzly to know if "such or such folder is in orgzly".

**Comment by artyhedgehog** [comment]
> I don't know the exact meaning of this sentence, but if you are looking for "the features wanted for this feature", yes I can give you some

No, I was actually thinking that if someone has better knowledge of Orgzly codebase, they could point me to the files in the code that currently responsible for parsing ID's from notes.

> I think that directories should be managed by orgzly. You have to add by hand all the folders, an option would be convenient to look for nested folders

Is it currently possible to do in Orgzly though? I know on "Notebooks" screen I can add a notebook (e.g. "my-notebook"), which will sync represent e.g. "my-notebook.org" file, but how do I connect a whole directory to Orgzly?
